protocol/x/affiliates/keeper/keeper.go

```diff
@@ -41,3 +49,154 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 }
 
 func (k Keeper) InitializeForGenesis(ctx sdk.Context) {}
+
+func (k Keeper) RegisterAffiliate(
+	ctx sdk.Context,
+	referee string,
+	affiliateAddr string,
+) error {
+	referredByPrefixStore := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.ReferredByKeyPrefix))
+	if referredByPrefixStore.Has([]byte(referee)) {
+		return errors.New("Referee already exists for address: " + referee)
```

Comment:
Use `errorsmod.Wrapf(` and a defined error code (instead of raw string inline) when returning errors. See [example](https://github.com/dydxprotocol/v4-chain/blob/372f645a4a546b123fc37e1209cb615b6eb62ddc/protocol/x/clob/keeper/process_operations.go#L220-L224)